lib/matplotlib/_types.py

```diff
@@ -0,0 +1,204 @@
+"""
+Concepts used by the matplotlib API that do not yet have a dedicated class.
+
+Matplotlib often uses simple data types like strings or tuples to define a
+concept; e.g. the line capstyle can be specified as one of 'butt', 'round',
+or 'projecting'. The classes in this module are used internally and document
+these concepts formally.
+
+As an end-user you will not use these classes directly, but only the values
+they define.
+"""
+
+from enum import Enum, auto
+from matplotlib import cbook, docstring
+
+
+class _AutoStringNameEnum(Enum):
+    """Automate the ``name = 'name'`` part of making a (str, Enum)."""
+    def _generate_next_value_(name, start, count, last_values):
+        return name
```

Comment:
May be worth noting that this will become unneeded (it is the default behavior) in Py3.10 per https://bugs.python.org/issue42385.